lib/components/WidthProvider.jsx

```diff
@@ -60,7 +60,7 @@ export default function WidthProvider<
       // eslint-disable-next-line
       const node = ReactDOM.findDOMNode(this); // Flow casts this to Text | Element
       if (node instanceof HTMLElement)
-        this.setState({ width: node.offsetWidth });
+        setTimeout(this.setState({ width: node.offsetWidth }), 0);
```

Comment:
I think making setTimeout on each event loop tick is not a good solution. Unfortunately, this change will negatively affect performance

Reply:
I believe a good rule of thumb is that anytime you use `setTimeout(() => ..., 0)` in a react component, is a sign that the solution needs to be re-engineered, or at the very least a significant code smell.